libwire/src/Sniffer.cpp

```diff
@@ -1,16 +1,23 @@
 #include "Sniffer.hpp"
 
 namespace Net {
-    Sniffer::Sniffer(PacketReader *packetReader) : _packetReader(packetReader) {}
- 
-    int Sniffer::startListening(InterfaceInfo const &interface) {
-        _packetReader->startListening(interface.getName());
-        return 0;
-    }
 
-    void Sniffer::notifyObserver() {
-        for(std::vector<SnifferObserver *>::iterator it = _observers.begin(); it != _observers.end(); ++it) {
-            (*it)->update();
-        }
+Sniffer::Sniffer(std::unique_ptr<PacketReader> packetReader)
+: _packetReader(std::move(packetReader)) {
```

Comment:
Take a unique pointer because the packet reader shoud only be stored in the Sniffer class,